src/scipp/core/data_group.py

```diff
@@ -233,28 +248,67 @@ def apply(self, func: Callable, *args, **kwargs) -> DataGroup:
         """Call func on all values and return new DataGroup containing the results."""
         return DataGroup({key: func(v, *args, **kwargs) for key, v in self.items()})
 
-    def apply_to_unsized(self, func: Callable, *args, **kwargs) -> DataGroup:
-        """
-        Call func on all "sized" values and return new DataGroup containing the results.
+    def _transform_dim(
+        self, func: Callable, *, dim: Union[None, str, Iterable[str]], **kwargs
+    ) -> DataGroup:
+        """Transform items that depend on one or more dimensions given by `dim`."""
+        dims = (dim,) if isinstance(dim, str) else dim
+
+        def intersects(item):
+            item_dims = _item_dims(item)
+            if dims is None:
+                return item_dims != ()
+            return set(dims).intersection(item_dims) != set()
 
-        "Sized" values are those that have a `dims` attribute, i.e., those that are
-        handled by __getitem__.
-        """
         return DataGroup(
             {
-                key: v if _item_dims(v) == () else func(v, *args, **kwargs)
+                key: v
+                if not intersects(v)
+                else operator.methodcaller(func, dim, **kwargs)(v)
                 for key, v in self.items()
             }
         )
 
+    def _reduce(
+        self, method: str, dim: Union[None, str, Tuple[str, ...]] = None, **kwargs
+    ) -> DataGroup:
+        reduce_all = operator.methodcaller(method, **kwargs)
+
+        def _reduce_child(v, dim):
+            if isinstance(v, (GroupByDataArray, GroupByDataset)):
+                child_dims = (dim,)
+            else:
+                child_dims = _item_dims(v)
+            # Reduction operations on binned data implicitly reduce over bin content.
+            # Therefore, a purely dimension-based logic is not sufficient to determine
+            # if the item has to be reduced or not.
+            binned = _is_binned(v)
+            if child_dims == () and not binned:
+                return v
+            if dim is None:
+                return reduce_all(v)
+            if isinstance(dim, str):
+                dims_to_reduce = dim if dim in child_dims else ()
```

Comment:
Should this be
```suggestion
                dims_to_reduce = (dim,) if dim in child_dims else ()
```
?
I.e. equivalent to processing the `dim` argument using `dims = (dim,) if isinstance(dim, str) else dim` as in `_transform_dim`.

Note that below, `if not dims_to_reduce` is also true for empty strings in the current implementation.